### ISSUE TITLE
Windows port: error from incorrect Win32 API error handling

### ISSUE THREAD

**Issue by sicklittlemonkey** [google/cups-connector#209]
Hi.

I was getting the following error:
Cloud Print Connector.error(1): [Job ...] Failed to submit to native print system: The parameter is incorrect.

This turned out to be from StartPage() which shares the same incorrect error checking as all the other Win32 APIs used. In short, you can't rely on GetLastError().

From [dll_windows.go](https://golang.org/src/syscall/dll_windows.go):
// The returned error is always non-nil, constructed from the result of GetLastError.
// **Callers must inspect the primary return value to decide whether an error occurred
// (according to the semantics of the specific function being called) before consulting
// the error.** The error will be guaranteed to contain syscall.Errno.

Here's the diff for a fix:
```
 func (hDC HDC) StartPage() error {
-	_, _, err := startPageProc.Call(uintptr(hDC))
-	if err != NO_ERROR {
+	r1, _, err := startPageProc.Call(uintptr(hDC))
+	if r1 <= 0 {
 		return err
 	}
 	return nil
 }
```
This does leave the door open for that classic Windows error "The operation completed successfully", but it's rare and always a bug in Windows. ; - )

**Note that because they aren't consistent each API needs to be checked as per its docs, though each API group tends to use the same style.** E.g. Spooler functions like GetPrinter/GetJob/SetJob would need "r1 == 0".

Hopefully I'll submit a pull request soon for my changes (I've added setting the username for the print job) but I won't update all the APIs as I'm not sure what approach you guys want to take.

Cheers,
Nick.

**Comment by sicklittlemonkey** [comment]
Ok, now I see some of the other Win32 API calls are handled as I did.

SetWorldTransform even handles the "The operation completed successfully" case:
```
	r1, _, err := setWorldTransformProc.Call(uintptr(hDC), uintptr(unsafe.Pointer(xform)))
	if r1 == 0 {
		if err == NO_ERROR {
			return fmt.Errorf("SetWorldTransform call failed; return value %d", int32(r1))
		}
		return err
	}
```
I've corrected all the error handling I could find that relied only on GetLastError() == NO_ERROR where a return code should be checked instead:
EnumPrinters
OpenPrinter
ClosePrinter
GetJob
SetJob
StartDoc
EndDoc
StartPage
EndPage
SetGraphicsMode

Cheers,
Nick.

**Comment by jacobmarble** [comment]
Nick, please do send a pull request. It looks like you've looked at this problem very closely.

**Comment by sicklittlemonkey** [comment]
I'm not very familiar with GitHub and made an attempt today that was quite the train wreck, sorry.

Hopefully it goes better tomorrow.

**Comment by jacobmarble** [comment]
:+1:

**Comment by sicklittlemonkey** [comment]
Fixed with merge #217.
